flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -219,52 +249,47 @@ public synchronized void stop() {
   @Override
   public Status process() throws EventDeliveryException {
     if (writer == null) {
-      try {
-        this.writer = newWriter(login, target);
-      } catch (DatasetException e) {
-        // DatasetException includes DatasetNotFoundException
-        throw new EventDeliveryException(
-          "Cannot write to " + getName(), e);
-      }
+      this.writer = newWriter();
     }
 
-    // handle file rolling
-    if ((System.currentTimeMillis() - lastRolledMs) / 1000 > rollIntervalS) {
-      // close the current writer and get a new one
-      writer.close();
-      this.writer = newWriter(login, target);
-      this.lastRolledMs = System.currentTimeMillis();
-      LOG.info("Rolled writer for " + getName());
+    if (shouldRoll()) {
```

Comment:
So, the only purpose of the above check that creates a new writer if the writer is null is to handle the case where the dataset doesn't exist when the sink starts, otherwise it will reset the rolled time and and not try to open a new writer until the roll time hits again.

I think the cleanest thing for me to do is to not reset the last rolled time unless a new writer was successfully created. That would remove the extra check of a null writer above. The only potential downside is that you'd probably end up with a weird looking log message when it logs that the writer was rolled when it starts up.